### PR TITLE
Update RFC process

### PR DESCRIPTION
**Pull request by davidmirror-ops**
## Describe your changes
Introduces the following changes to the RFC process:

1. Keep the Lazy Consensus approach for decision making but giving the Steering Committee a binding voting role
2. Introduce the Final Comment Period stage
3. Define scope of a RFC
4. Define acceptance criteria
5. Eliminate the need to start on Markdown and continue on PR
6. Incline to async communication

## Check all the applicable boxes 

- [X] I updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] All commits are signed-off.

### CODE REVIEW

File: rfc/README.md [hunk 1]
```diff
@@ -25,111 +22,64 @@ Changes not requiring RFCs include:
...
-
-## The process
 
-[![](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVERcbiAgICBBW1N0YXJ0XSAtLT58V3JpdGUgUkZDfCBCKE9wZW4gYSBQUilcbiAgICBCIC0tPiBDe1BSIFJldmlld31cbiAgICBDIC0tPnxBcHByb3ZlZHwgRFtNZXJnZV1cbiAgICBDIC0tPnxGZWVkYmFja3wgQ1xuICAgIEMgLS0-fFJlamVjdGVkfCBGW1BSIENsb3NlZF0iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9LCJ1cGRhdGVFZGl0b3IiOmZhbHNlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6ZmFsc2V9)](https://mermaid-js.github.io/mermaid-live-editor/edit##eyJjb2RlIjoiZ3JhcGggVERcbiAgICBBW1N0YXJ0XSAtLT58V3JpdGUgUkZDfCBCKE9wZW4gYSBQUilcbiAgICBCIC0tPiBDe1BSIFJldmlld31cbiAgICBDIC0tPnxBcHByb3ZlZHwgRFtNZXJnZV1cbiAgICBDIC0tPnxGZWVkYmFja3wgQ1xuICAgIEMgLS0-fFJlamVjdGVkfCBGW1BSIENsb3NlXSIsIm1lcm1haWQiOiJ7XG4gIFwidGhlbWVcIjogXCJkZWZhdWx0XCJcbn0iLCJ1cGRhdGVFZGl0b3IiOmZhbHNlLCJhdXRvU3luYyI6dHJ1ZSwidXBkYXRlRGlhZ3JhbSI6ZmFsc2V9)
+- Gathering feedback from team or community members first, to confirm that the changes will indeed be useful
+- Starting a Discussion at the [RFC Incubator](https://github.com/flyteorg/flyte/discussions/new?category=rfc-incubator) to gauge interest.
```
[the remark this answers is not in the document]
Comment (davidmirror-ops) [reply rfc/README.md]: added some lines, wdyt?

**Review by cosmicBboy — DISMISSED** [review]
This is great!

**Review by bstadlbauer — APPROVED** [review]
Looks good to me - thank you!

File: rfc/README.md [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
 ## When you need to follow this process
-[When you need to follow this process]: #when-you-need-to-follow-this-process
-
-Significant changes to the system or changes to the user experience would require an RFC. Examples of changes can be:
+[When you need to follow this process]: #when-you-need-to-follow-this-process 
+You need to follow this process if you intend to make a significant change to the system or changes to the user experience. What constitutes a "significant" change is evolving based on community feedback, but may include the following:
 
   - Semantics or syntax changes, other than bugfixes (e.g. introducing a new Flytekit language feature)
-  - Core-Language updates (e.g. FlyteIdl changes that introuce new wire-format)
+  - Core-Language updates (e.g. FlyteIdl changes that introduce new wire-format)
```
Comment (fg91) [inline rfc/README.md:12]:
```suggestion
  - Core-Language updates (e.g. FlyteIdl changes that introduce a new wire-format)
```

File: rfc/README.md [hunk 3]
```diff
@@ -25,111 +22,67 @@ Changes not requiring RFCs include:
...
+- Gathering feedback from team or community members first, to confirm that the changes will indeed be useful
+- Starting a Discussion at the [RFC Incubator](https://github.com/flyteorg/flyte/discussions/new?category=rfc-incubator) to gauge interest. Once you have received positive feedback, especially from Maintainers or Steering Committee members, please proceed to:
+    - Ask in the Discussion for confirmation to submit an RFC
+    - If there's no objection (silence is approval) then create an Issue from the Discussion ([see how](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue#creating-an-issue-from-discussion))
+    - Proceed to [open a PR](#opening-a-pull-request)
```
Comment (fg91) [inline rfc/README.md:36]: Would be nice to link `### Opening a Pull Request` here.

File: rfc/README.md [hunk 4]
```diff
@@ -25,111 +22,67 @@ Changes not requiring RFCs include:
...
+- Discussing the topic on the [#contribute](https://flyte-org.slack.com/archives/C04NJPLRWUX) Slack channel
+- Adding the topic to the Contributor's [meeting agenda](https://hackmd.io/@davidmirror/rkqCpbK1n) to make sure it alligns with roadmap
+- Taking the time to produce a well-written, well-thought-of document by using the template located [here](https://github.com/flyteorg/flyte/blob/RFC-Process/rfc/RFC-0000-Template.md).
 
-### The bare-minimum
+## Where to store RFCs
 
-RFCs must be introduced as a `.md` (Markdown-format) file in a PR into the flyte repo. Details of the location to store them can be [found below](#Where-to-store-RFCs).
+The Flyte repo on GitHub has an RFC folder with 3 directories:
+- **Core language:** proposals to `FlyteIdl` that change the wire-format in any way are considered significant changes that require revision and approval.
```
Comment (fg91) [inline rfc/README.md:44]: An example of what change to flyteidl constitutes or doesn't constitute a change of the wire-format.

File: rfc/README.md [hunk 5]
```diff
@@ -25,111 +22,67 @@ Changes not requiring RFCs include:
...
-  Reviewer: At least one of [flyte maintainers](https://github.com/orgs/flyteorg/teams/flyte-maintainers/members).
-  
-## Reviewing RFCs
-
-Once the RFC is created in a PR, reviewers can take up to 3 weeks to review, test, and either approve or recommend adjustments. They may call for meetings, demos, or other measures deemed necessary.
+Once a pull request is opened, the RFC is now in development and the following will happen:
 
-### Instructions for Reviewers
+* It will be introduced in a future Contributor's meetup, happenning every other week, except othwerwise informed.
+* The proposal will be discussed as much as possible in the RFC pull request directly. Any outside discussion will be summarized in the comment thread
```
Comment (fg91) [inline rfc/README.md:67]:
```suggestion
* The proposal will be discussed as much as possible in the RFC pull request directly. Any outside discussion will be summarized in the comment thread.
```

File: rfc/README.md [hunk 6]
```diff
@@ -25,111 +22,67 @@ Changes not requiring RFCs include:
...
+* When deemed "ready", a maintainer or TSC member will propose a "motion for Final Comment Period (FCP)" along with a disposition of the outcome (merge, close, or postpone). This step is taken when enough discussions of the tradeoffs have taken place and the community is in a position to make a decision. 
+* The proposal enters FCP unless there's any objection (lazy consensus)
```
Comment (fg91) [inline rfc/README.md:69]:
```suggestion
* The proposal enters FCP unless there's any objection (lazy consensus).
```

File: rfc/README.md [hunk 7]
```diff
@@ -25,111 +22,67 @@ Changes not requiring RFCs include:
...
+* The Final Comment Period will last 7 days. If there's no objection, the FCP can close early
+* If no substantial new arguments or ideas are raised, the FCP will follow the outcome decided
```
Comment (fg91) [inline rfc/README.md:71]:
```suggestion
* If no substantial new arguments or ideas are raised, the FCP will follow the outcome decided.
```

**Review by fg91 — DISMISSED** [review]
Can the first version of the process diagram be removed? Only the second appears to be used in the text.

Apart from that I left minor comments and suggestions, feel free to address or ignore as you see fit.

Thanks for refining this, LGTM.

File: rfc/RFC-Process-diagram.png
[the remark this answers is not in the document]
Comment (davidmirror-ops) [reply rfc/RFC-Process-diagram.png:1]: Removed

File: rfc/README.md (hunk 4 above)
Comment (davidmirror-ops) [reply rfc/README.md:44]: Great question. @kumare3 / @cosmicBboy do you have an idea?

**Review by cosmicBboy — APPROVED** [review]

**Pull request merged**